### ISSUE TITLE
How to use efficientNet2 as backbone for efficientdet?

### ISSUE THREAD

**Issue by drahmad89** [google/automl#1004]

**Comment by moaljazaery** [comment]
Interested

**Comment by kartik4949** [comment]
Hi,
The idea efficientNetV2 is centered around is progressive learning and also ( architectural changes), hence by just changing backbone won't be enough, progressive image learning should be added in training script.
but im not sure, if that will help  in EfficientDet.
@mingxingtan can explain better here. :)

**Comment by mingxingtan** [comment]
Yeah, you should be easily plugin efficientnetv2 to efficientdet by changing this line: https://github.com/google/automl/blob/d07468a28df7100251332865332f2412eafe6f31/efficientdet/keras/efficientdet_keras.py#L805

Hopefully I will get TPU resources soon, and retrain those detection models with effnetv2.

**Comment by kartik4949** [comment]
> Yeah, you should be easily plugin efficientnetv2 to efficientdet by changing this line: https://github.com/google/automl/blob/d07468a28df7100251332865332f2412eafe6f31/efficientdet/keras/efficientdet_keras.py#L805
> 
> 
> 
> Hopefully I will get TPU resources soon, and retrain those detection models with effnetv2.

Progressive learning with detection models too?

**Comment by mingxingtan** [comment]
Oh, no progressive learning, I meant just the backbone.

Never tried progressive learning for detection, but in general, the anchor boxes might be the issue.

**Comment by drahmad89** [comment]
The idea for me is to benefit from the speedup in the fusion layers. I was going through a customized TVM implementation to apply what V2 has already done!

**Comment by kartik4949** [comment]
> Oh, no progressive learning, I meant just the backbone.
> 
> 
> 
> Never tried progressive learning for detection, but in general, the anchor boxes might be the issue.

Oh cool 
Gotcha :)

**Comment by drahmad89** [comment]
> > Yeah, you should be easily plugin efficientnetv2 to efficientdet by changing this line: https://github.com/google/automl/blob/d07468a28df7100251332865332f2412eafe6f31/efficientdet/keras/efficientdet_keras.py#L805
> > 
> > Hopefully I will get TPU resources soon, and retrain those detection models with effnetv2.
> 
> Progressive learning with detection models too?



> Yeah, you should be easily plugin efficientnetv2 to efficientdet by changing this line:
> 
> https://github.com/google/automl/blob/d07468a28df7100251332865332f2412eafe6f31/efficientdet/keras/efficientdet_keras.py#L805
> 
> Hopefully I will get TPU resources soon, and retrain those detection models


Sorry i followed this line and doesn’t require to make changes in efficientnet builder? And provide the backbone weights?

**Comment by akashAD98** [comment]
@mingxingtan can you please explain detailed steps for training our custom data using efficient netv2.it would be a great help. Thanks in advance
